src/Twig/Extension/TemplateRegistryExtension.php

```diff
@@ -11,21 +11,29 @@
 
 namespace Sonata\AdminBundle\Twig\Extension;
 
-use Sonata\AdminBundle\Admin\AdminInterface;
-use Sonata\AdminBundle\Admin\Pool;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\ContainerInterface;
+use Sonata\AdminBundle\Admin\TemplateRegistryInterface;
+use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
 use Twig\Extension\AbstractExtension;
 use Twig\TwigFunction;
 
 final class TemplateRegistryExtension extends AbstractExtension
 {
     /**
-     * @var Pool
+     * @var TemplateRegistryInterface
      */
-    private $pool;
+    private $generalTemplateRegistry;
 
-    public function __construct(Pool $pool)
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    public function __construct(TemplateRegistryInterface $generalTemplateRegistry, ContainerInterface $container)
     {
-        $this->pool = $pool;
+        $this->generalTemplateRegistry = $generalTemplateRegistry;
+        $this->container = $container;
```

Comment:
You can and probably should bump the sf dependency to 3.4, to allow a service locator here. We wanted to do it, and were just waiting for a good reason to do so. I think this would be it.

Reply:
I'd prefer to do this in a separate PR. I found another place that also needs it, so I could do both at the same time that way. Is that acceptable?